### PR TITLE
Added available seats to drivers list, modified profile components

### PR DESCRIPTION
**Pull request by nancy-yuan**
Refactored drivers' list to remove direct DOM manipulation in ts file.
Added available and total seats to drivers list.
Changed location from default "Morgantown, WV" to current user's home address.
Fixed bugs in profile contact, profile location, profile membership, and profile car components.
Added user input validation to profile components and converted to reactive forms.
Added address validation to profile location component.
Added success and error messages to profile components.

### CODE REVIEW

File: src/app/components/profile-location/profile-location.component.ts [hunk 1]
```diff
@@ -12,82 +12,120 @@ import { ValidationService } from 'src/app/services/validation-service/validatio
 })
 export class ProfileLocationComponent implements OnInit {
 
+  currentUser;
+  
   states = ['AL','AK','AZ','AR','CA','CO','CT','DE','FL','GA','HI','ID','IL','IN','IA','KS',
```
Comment (flacks) [inline src/app/components/profile-location/profile-location.component.ts]: Why is this unsorted 😢

Comment (flacks) [inline src/app/components/profile-location/profile-location.component.ts]: Missing 'DC'

File: src/app/components/driver-list/driver-list.component.ts [hunk 2]
```diff
@@ -17,92 +18,89 @@ import { environment } from '../../../environments/environment';
...
-      this.showDriversOnMap(this.location, this.drivers);
-    });
+    this.currentUserID = +sessionStorage.getItem('userid');
+
+    // sets the location to the driver's home address
+    this.userService.getDriverById(this.currentUserID).subscribe(
+      (currentUser) => {
+
+        let currentAddress = currentUser.homeAddress;
+        this.location = currentAddress.street + ", " + currentAddress.city + ", " + currentAddress.state;
```
Comment (Jcfo233) [inline src/app/components/driver-list/driver-list.component.ts:45]: This leaves out the apartment/address 2 value in address

File: src/app/components/driver-list/driver-list.component.ts [hunk 3]
```diff
@@ -17,75 +18,89 @@ import { environment } from '../../../environments/environment';
...
+
+        this.getGoogleApi();
+
+        this.userService.getRidersForLocation1(this.location).subscribe(
+          (res) => {
```
Comment (pjlawrence) [inline src/app/components/driver-list/driver-list.component.ts]: Can we be more descriptive than "res" here. Maybe "user" if a user is being returned.

File: src/app/components/driver-list/driver-list.component.ts [hunk 4]
```diff
@@ -119,62 +134,53 @@ displayRoute(origin, destination, service, display) {
...
-      var service = new google.maps.DistanceMatrixService;
+      var availableSeats: number;
+      var totalSeats: number;
+
+      this.carService.getCarTripByUserId(element.id).subscribe(
+        (carTrip) => {
+          // @ts-ignore
+          availableSeats = carTrip.currentTrip.availableSeats;
+          // @ts-ignore
+          totalSeats = carTrip.car.seats;
```
Comment (pjlawrence) [inline src/app/components/driver-list/driver-list.component.ts]: Why do we need these ignore lines?

File: src/app/components/driver-list/driver-list.component.ts [hunk 5]
```diff
@@ -116,11 +114,11 @@ getGoogleApi()  {
  */
 displayRoute(origin, destination, service, display) {
     service.route({
-      origin: origin,
-      destination: destination,
+      origin,
+      destination,
       travelMode: 'DRIVING',
       //avoidTolls: true
```
Comment (Jcfo233) [inline src/app/components/driver-list/driver-list.component.ts]: Commented out line

File: src/app/components/profile-location/profile-location.component.css [hunk 6]
```diff
@@ -3,11 +3,28 @@
...
+    font-weight:bold;
+    text-align:left;
+}
+
+
+/* error */
+.error {
+    margin-left:-8px;
+    width:320px !important;
+    padding-left:0px !important;
```
Comment (pjlawrence) [inline src/app/components/profile-location/profile-location.component.css:22]: why are you using !important. This should be avoided almost always. Have you tried adding greater specificity? Just making it "html .error" will make this command more important and making it "html body .error" even more so.

File: src/app/components/profile-car/profile-car.component.ts [hunk 7]
```diff
@@ -9,38 +10,54 @@ import { Car } from 'src/app/models/car';
...
+  ngOnInit() {}
+
   /**
    * Alters the fields of the car intialized in ngOnInit() to match those given by this component.
    */
   updatesCarInfo(){
-    this.currentCar.make = this.make;
-    this.currentCar.model= this.model;
-    this.currentCar.seats = this.nrSeats;
-    //console.log(this.currentUser);
```
Comment (Jcfo233) [inline src/app/components/profile-car/profile-car.component.ts:41]: commented out line

File: src/app/components/profile-location/profile-location.component.css (hunk 6 above)
Comment (pjlawrence) [reply src/app/components/profile-location/profile-location.component.css:22]: (of course using html elements that are inherently related to where your style is used is a better practice than using top level tags)

File: src/app/services/car-service/car.service.ts [hunk 8]
```diff
@@ -56,13 +56,31 @@ export class CarService {
...
+		return this.http.put(`${this.url}${car.carId}`, car).toPromise();
 	}
 
-
+	/**
+	 * Identical to the above method except that it returns an observable.
+	 * @param userId 
+	 */
+	updateCarInfo2(car: Car): Observable<Car> {
+		//console.log(user);
```
Comment (Jcfo233) [inline src/app/services/car-service/car.service.ts]: commented out line

File: src/app/services/car-service/car.service.ts [hunk 9]
```diff
@@ -56,13 +56,31 @@ export class CarService {
...
+	getCarTripByUserId(userId: number) {
+		return this.http.get(`${this.url}trips/driver/${userId}`)
+	}
 
+	/**
+	 * Updates information on a car
+	 * @param car 
+	 */
 	updateCarInfo(car: Car) {
 		//console.log(user);
```
Comment (Jcfo233) [inline src/app/services/car-service/car.service.ts]: commented out line

File: src/app/services/user-service/user.service.ts [hunk 10]
```diff
@@ -68,12 +68,11 @@ export class UserService {
 	 * Identical to the above, except that it does not give a promise.
 	 * @param idParam2 
 	 */
-	getUserById2(idParam2: string): Observable<User>{
-		
+	getUserById2(idParam2: String): Observable<User>{
+
 		//console.log(this.url)
```
Comment (Jcfo233) [inline src/app/services/user-service/user.service.ts]: commented out line

File: src/app/services/user-service/user.service.ts [hunk 11]
```diff
@@ -42,21 +42,21 @@ export class UserService {
...
-	
+
 	/**
 	 * A GET method to retrieve one user from the database with the given id. Return a promise.
 	 * @param idParam 
 	 */
 	getUserById(idParam: number){
-		
+
 		console.log(this.url)
```
Comment (Jcfo233) [inline src/app/services/user-service/user.service.ts]: is this log intentional?

**Review by IncarnateGaming — COMMENTED** [review]
My review is mostly on coding best practices. I'm not familiar enough with our front end to do a proper integration review, so you will need others as well. Overall very nicely done.

File: src/app/components/driver-list/driver-list.component.ts (hunk 5 above)
Comment (kfilio) [reply src/app/components/driver-list/driver-list.component.ts]: should remove commented out lines from legacy code?
this one in particular came with the code

**Review by flacks — APPROVED** [review]
Apart from dead comments, LGTM

File: src/app/components/driver-list/driver-list.component.ts (hunk 5 above)
Comment (Michael-Kochis) [reply src/app/components/driver-list/driver-list.component.ts]: I would say yes, remove any legacy commented code.  We aren't going to use it, and it's just a code smell.

File: src/app/components/driver-list/driver-list.component.ts (hunk 2 above)
Comment (kfilio) [reply src/app/components/driver-list/driver-list.component.ts:45]: currentAddress.street also includes the apt number

File: src/app/components/profile-location/profile-location.component.css (hunk 6 above)
Comment (kfilio) [reply src/app/components/profile-location/profile-location.component.css:22]: made attempts to increase specificity but it just will not cooperate. will have to leave them in for now

File: src/app/components/profile-location/profile-location.component.ts (hunk 1 above)
Comment (kfilio) [reply src/app/components/profile-location/profile-location.component.ts]: sorted based on *~full name~* abc order

Comment (kfilio) [reply src/app/components/profile-location/profile-location.component.ts]: potentially very dangerous because of address validation

Comment (flacks) [reply src/app/components/profile-location/profile-location.component.ts]: 🥇

**Review by Jcfo233 — APPROVED** [review]

**Pull request merged**